### PR TITLE
[SPARK-25200][YARN] Allow specifying HADOOP_CONF_DIR as spark property

### PR DESCRIPTION
**Pull request by adambalogh**
SPARK-25200 - https://github.com/apache/spark/pull/22289

Replicates the behaviour of setting `HADOOP_CONF_DIR`, but as a spark config `spark.yarn.conf.dir`

### CODE REVIEW

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 1]
```diff
@@ -694,24 +694,21 @@ private[spark] class Client(
...
-            logWarning("Failed to list files under directory " + dir)
-          } else {
-            files.foreach { file =>
-              if (file.isFile && !hadoopConfFiles.contains(file.getName())) {
-                hadoopConfFiles(file.getName()) = file
-              }
+    val confDirsEnvKeys = Seq("HADOOP_CONF_DIR", "YARN_CONF_DIR")
+    val confDirProp = sparkConf.getOption("spark.yarn.conf.dir")
+
+    val confDirPaths = (confDirsEnvKeys.map(sys.env.get) :+ confDirProp).flatMap(_.toList)
```
Comment (robert3005) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: can just do `.flatten` here

**Comment by robert3005** [comment]
Going to repush to this repo so that tests run

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala (hunk 1 above)
Comment (adambalogh) [reply resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: fixed

**Comment by adambalogh** [comment]
Thanks for the swift review @robert3005 ! I made the changes

**Pull request closed without merging**